environments/java_codegen_monitoring/nginx.conf

```diff
@@ -6,12 +6,17 @@ events {
 }
 
 http {
+    log_format  main  '$remote_addr - $remote_user [$time_local] [$request] status: $status request_time: $request_time  '
+        'bytes: $body_bytes_sent upstream: [$upstream_addr]';
+    access_log /dev/stdout main;
+    error_log /dev/stdout;
+
     server {
         listen 9001;
         large_client_header_buffers 4 32k;
         sendfile            on;
         keepalive_timeout   65;
-        client_max_body_size 20M;
+        client_max_body_size 1M;
```

Comment:
Is this big enough? Where did we come up with this number? What does PPS server support?

Reply:
for a single request I tried multiple combination 100k, 500k, 1MB, 1,4MB which works fine. Above 2MB we have this retry problem, SAP is retrying automatically if request does not complete under some time (I did not find any documentation with this time specifically but is around 60 sec)

Internally we have drum predictor that is able to do predictions quick, but the monitoring part takes longer since it uses spooler API. We can do multiple improvements to able to support larger payloads, but we will still have a limit based on by SAP proxy.